internal/datasourcev2/streaming_data_source.go

```diff
@@ -275,14 +276,18 @@ func (sp *StreamProcessor) consumeStream(stream *es.Stream, closeWhenReady chan<
 					break
 				}
 
-				switch currentChangeSet.intent.Payloads[0].Code {
-				case fdv2proto.IntentTransferFull:
-					{
-						sp.dataDestination.SetBasis(updates, selector, true)
-						sp.setInitializedAndNotifyClient(true, closeWhenReady)
-					}
-				case fdv2proto.IntentTransferChanges:
+				if currentChangeSet.intent == nil {
```

Comment:
Only the initial payload has an intent. Subsequent updates are just assumed to be changesets.